schema/sqlite/setup.go

```diff
@@ -133,17 +134,48 @@ func CreateNamespaces(cfg *config.SQL, namespaces ...*NamespaceConfig) error {
 // the namespace via the CreateNamespaces function.
 //
 // Note: this function may receive breaking changes or be removed in the future.
-func NewNamespaceConfig(activeClusterName, namespace string, global bool) *NamespaceConfig {
+func NewNamespaceConfig(
+	activeClusterName string,
+	namespace string,
+	global bool,
+	customSearchAttributes map[string]enumspb.IndexedValueType,
+) (*NamespaceConfig, error) {
+	dbCustomSearchAttributes := searchattribute.GetSqlDbIndexSearchAttributes().CustomSearchAttributes
+	fieldToAliasMap := map[string]string{}
+	for saName, saType := range customSearchAttributes {
+		targetFieldName := ""
+		cntUsed := 0
+		for fieldName, fieldType := range dbCustomSearchAttributes {
+			if fieldType != saType {
+				continue
+			}
+			if _, ok := fieldToAliasMap[fieldName]; !ok {
+				targetFieldName = fieldName
+				break
+			}
+			cntUsed++
+		}
+		if targetFieldName == "" {
+			return nil, fmt.Errorf(
+				"cannot have more than %d search attributes of type %s",
+				cntUsed,
+				saType,
+			)
+		}
```

Comment:
This sounds like the field was used too often, but isn't the condition checking if the field was _never_ used?

Reply:
There's a limit of the number of search attributes you can create.

Reply:
So `targetFieldName` is initialized to an empty string at the start. When none of the fields the user provided match (ie `fieldToAliasMap` is empty), then the user will see an error message saying they have _too many_ search attributes defined - when they actually have defined none?

Reply:
This accounts for the user trying to add say 5 search attributes of type keyword at the same time, but there's only 3 slots available.

Reply:
Okay, I re-read it and agree!